src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1177,44 +1279,31 @@ private sealed class LoaderHolder
 
                 // Make sure the loader binds to us.
                 var info = host.ComponentCatalog.GetLoadableClassInfo<SignatureDataLoader>(loader.Name);
-                if (info.Type != typeof(IDataLoader) || info.ArgType != typeof(Arguments))
+                if (info.Type != typeof(IDataLoader) || info.ArgType != typeof(Options))
                     goto LDone;
 
-                var argsNew = new Arguments();
-                // Copy the non-core arguments to the new args (we already know that all the core arguments are default).
-                var parsed = CmdParser.ParseArguments(host, CmdParser.GetSettings(host, args, new Arguments()), argsNew);
-                ch.Assert(parsed);
-                // Copy the core arguments to the new args.
-                if (!CmdParser.ParseArguments(host, loader.GetSettingsString(), argsNew, typeof(ArgumentsCore), msg => ch.Error(msg)))
+                var optionsNew = new Options();
+                // Copy the core arguments to the new options.
+                if (!CmdParser.ParseArguments(host, loader.GetSettingsString(), optionsNew, typeof(Options), msg => ch.Error(msg)))
                     goto LDone;
 
-                cols = argsNew.Columns;
+                // Set the non core arguments in the new options.
+                optionsNew.UseThreads = options.UseThreads;
+                optionsNew.HeaderFile = options.HeaderFile;
+                optionsNew.MaxRows = options.MaxRows;
+
+                cols = optionsNew.Columns;
                 if (Utils.Size(cols) == 0)
                     goto LDone;
 
                 error = false;
-                args = argsNew;
+                options = optionsNew;
 
             LDone:
                 return !error;
             }
         }
 
-        /// <summary>
-        /// Checks whether the source contains the valid TextLoader.Arguments depiction.
-        /// </summary>
-        public static bool FileContainsValidSchema(IHostEnvironment env, IMultiStreamSource files, out Arguments args)
-        {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(files, nameof(files));
-            args = new Arguments();
-            Column[] cols;
-            bool error;
-            bool found = TryParseSchema(h, files, ref args, out cols, out error);
-            return found && !error && args.IsValid();
-        }
-
```

Comment:
is this deletion intentional?

Reply:
Well I noticed we never call it, so I decided to remove it, but maybe would be better to make it internal?

---
In reply to: [254402153](https://github.com/dotnet/machinelearning/pull/2417#discussion_r254402153) [](ancestors = 254402153)

Reply:
Since this might be used in TLC, I am just going to make it internal and keep it.

---
In reply to: [254512295](https://github.com/dotnet/machinelearning/pull/2417#discussion_r254512295) [](ancestors = 254512295,254402153)